webapp/lib/view.dart

```diff
@@ -535,13 +504,27 @@ class BatchRepliesConfigurationView extends PackageConfiguratorView {
     );
   }
 
-  void _addTagDropdown(MouseEvent event, List<String> tags) {
-    var tagsList = (event.target as Element).parent;
-    if (tagsList.children.last.classes.contains('add-tag-dropdown')) tagsList.lastChild.remove();
+  SpanElement _addTagAction(List<String> tags) {
+    return new SpanElement()
+      ..classes.add('tags__actions')
+      ..append(
+        new ButtonElement()
+          ..classes.add('button-add-tag')
+          ..text = '+'
+          ..onClick.listen((event) {
+            _addTagDropdown(event, model.tags);
+            event.stopPropagation();
+          })
+      );
+  }
 
+  void _addTagDropdown(MouseEvent event, List<String> tags) {
```

Comment:
in general I'd prefer it if we can avoid passing `events` around if we're not using them, and instead use explicit points to objects, esp. if we've just created them. This is because the event doesn't just get triggered on the element you set it to, but instead on the children elements and if they don't handle it (and stop propagation), it then bubbles up through the parents. This means that, depending on where you click, you might get different results when we would do `.parent` below.

so my suggestion would be that you keep hold of the parent `SpanElement` created in `_addTagAction` and then pass that directly to `_addTagDropdown`.

Reply:
passed them as references